demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py

```diff
@@ -602,6 +607,127 @@ def verify_test_data_exists(test_data_path: Path) -> Tuple[List[str], List[str]]
     return missing_event_data, missing_expected_values_data
 
 
+def verify_event_id_does_not_exist_on_tenant(
+    xsiam_client: XsiamApiClient,
+    mr: ModelingRule,
+    test_data: init_test_data.TestData,
+):
+    """
+    Verify that the event ID does not exist on the tenant.
+    Args:
+        xsiam_client (XsiamApiClient): Xsiam API client.
+        mr (ModelingRule): Modeling rule object parsed from the modeling rule file.
+        test_data (init_test_data.TestData): Test data object parsed from the test data file.
+    """
+    logger.debug("Verifying that the event ID does not exist on the tenant.")
+    for rule in mr.rules:
+        try:
+            query = build_event_id_query(rule, test_data)
+            run_query(xsiam_client, query)
+        except requests.exceptions.HTTPError:
+            logger.error(
+                (
+                    "[red]Error executing XQL query, potential reasons could be:\n - mismatch between "
+                    "dataset/vendor/product marked in the test data from what is in the modeling rule\n"
+                    " - dataset was not created in the tenant\n - model fields in the query are invalid\n"
+                    "Try manually querying your tenant to discover the exact problem.[/red]"
+                ),
+                extra={"markup": True},
+            )
+        except XsiamApiQueryError:
+            # If the query fails, it means that the event ID does not exist on the tenant.
+            logger.info(
+                "[green]The event ids does not exists on the tenant[/green]",
+                extra={"markup": True},
+            )
+            return
+    logger.error(
+        ("[red]The event id already exists in the tenant[/red]"),
+        extra={"markup": True},
+    )
+    raise typer.Exit(1)
```

Comment:
Why fail the test on it? 
In the previous PR this was not like this?
Where do you replace the existing uuid with new one?